### DISCUSSION TITLE
Popen executable fails to find dll, need help to wrap it in SetDllDirectory calls

### DISCUSSION THREAD

**Discussion by ericoporto** [pyinstaller/pyinstaller#8125] [Help] [answered]
I need to use Popen from a onefile executable on Windows and am affected by  #3795

I think I can just call SetDllDirectory (NULL) before running the child process and then run SetDllDirectory (extraction_path) after, but there isn't a clean way to do so - that I can figure it out. I can't even figure it out how to get the extraction path. Essentially I would like to SetDllDirectory to NULL, call Popen and later after Popen is finished SetDllDirectory back to what it was. My Popen call is made in a way it's blocking because it's a command line app.

Ah, curious thing, my command line app has a graphical version that is also built with pyinstaller but it's apparently not affected by the issue!

**Comment by rokm** [comment, marked as answer]
> I can't even figure it out how to get the extraction path

It is stored in `sys._MEIPASS`.

If your application does not have any delayed module imports, you could probably also just call ` SetDllDirectory(NULL)` once, after importing all the modules that your application uses (i.e., you might get away withotu restoring it back).

**Reply by ericoporto** [reply]
Thanks! I will use that and see if it works :)

**Reply by ericoporto** [reply]
Sorry @rokm , I get `Unresolved reference 'NULL' `, what is the way to do this? Edit: oh, right, using `None` fixes it!
